hotsos/core/root_manager.py

```diff
@@ -0,0 +1,112 @@
+import os
+import shutil
+import sys
+import tarfile
+import tempfile
+from functools import cached_property
+
+from hotsos.core.config import HotSOSConfig
+from hotsos.core.host_helpers import CLIHelper
+
+
+class DataRootManager(object):
+    TYPE_HOST = 0
+    TYPE_SOSREPORT = 1
+
+    def __init__(self, path, sos_unpack_dir=None):
+        self.path = path
+        self.sos_unpack_dir = sos_unpack_dir
+
+    @cached_property
+    def tmpdir(self):
+        if self.sos_unpack_dir is None:
+            return tempfile.mkdtemp()
+
+        return self.sos_unpack_dir
```

Comment:
I couldn't tell if you checked this existed? option help said it should so figured there would be a graceful failure?

Reply:
yeah actually python-click checks that - https://github.com/canonical/hotsos/blob/c2d0d9bbabe75f3caa99f521e1508f0079e2dfbd/hotsos/cli.py#L260

Reply:
so basically it will fail immediately if you provide a path that doesnt exist

Reply:
So is that needed for sos_unpack_dir too? understand it is friday and my brain is mush :)